update provisioning state info (#1032)
metal3-io/baremetal-operator: docs/api.md

* *state* -- The current state of any ongoing provisioning operation.
  The following are the currently supported ones:
  * *\<empty string\>* -- There is no provisioning happening, at the moment.

>   * *unmanaged* -- There is insufficient information available to register the host.

Reviewer: ```suggestion
  * *unmanaged* -- There is an insufficient information available to register the host.
```

Author: updated, PTAL.

  * *registering* -- The host's BMC details are being checked.
  * *match profile* -- The discovered hardware details on the host
    are being compared against known profiles.
  * *available* -- The host is available to be consumed. (This state was
    previously known as *ready*.)
  * *preparing* -- The existing configuration will be removed, and the new
    configuration will be set on the host.
  * *provisioning* -- An image is being written to the host's disk(s).
  * *provisioned* -- An image has been completely written to the host's
    disk(s).
  * *externally provisioned* -- Metal³ does not manage the image on the host.
  * *deprovisioning* -- The image is being wiped from the host's disk(s).
  * *inspecting* -- The hardware details for the host are being collected
    by an agent.

>   * *deleting* -- The host ready for deletion is cleaning up.

Reviewer: maybe:
```suggestion
  * *deleting* -- The host is being deleted from the cluster.
```
